src/main/java/redis/clients/jedis/util/CaffeineCSC.java

```diff
@@ -0,0 +1,50 @@
+package redis.clients.jedis.util;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.util.concurrent.TimeUnit;
+import redis.clients.jedis.ClientSideCache;
+
+public class CaffeineCSC extends ClientSideCache {
```

Comment:
Just a thought that comes to my mind when I read this: the general advice is to favor composition over inheritance. Maybe we could model this via composition, put an interface on top of `CaffeineCSC` and `GuavaCSC` and inject that interface into `ClientSideCache`?

Reply:
ClientSideCache itself is that interface (abstract class). It is an optional parameter for our actual client classes.